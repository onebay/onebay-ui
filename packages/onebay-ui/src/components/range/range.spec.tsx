import { mount } from '@vue/test-utils'
import Range from './index'

describe('Range Snap', () => {
  it('render initial Range', () => {
    const wrapper = mount(<Range />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props className', () => {
    const wrapper = mount(<Range className='test' />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props style', () => {
    const wrapper = mount(<Range style='color:red;' />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props sliderStyle', () => {
    const wrapper = mount(<Range sliderStyle='color:red;' />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props railStyle', () => {
    const wrapper = mount(<Range railStyle='color:red;' />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props blockSize', () => {
    const wrapper = mount(<Range blockSize={3} />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props trackStyle', () => {
    const wrapper = mount(<Range railStyle='color:red;' />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props value', () => {
    const wrapper = mount(<Range value={[0, 30]} />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props min', () => {
    const wrapper = mount(<Range min={10} value={[10, 30]} />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props max', () => {
    const wrapper = mount(<Range max={50} value={[10, 30]} />)
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render Range -- props disabled', () => {
    const wrapper = mount(<Range disabled />)
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('Range Event', () => {
  it('Rate onChange Event disabled', () => {
    const wrapper = mount(<Range disabled />)
    wrapper.find('.ob-range__slider').trigger('touchmove', { x: 50 })
    wrapper.find('.ob-range__slider').trigger('touchend', { x: 100 })
  })

  it('Rate onChange Event', () => {
    const wrapper = mount(<Range />)
    wrapper.find('.ob-range__slider').trigger('touchmove', { x: 50, touches: [{ clientX: 50 }] })
    wrapper.find('.ob-range__slider').trigger('touchend', { x: 100 })
    expect(wrapper.emitted()).toHaveProperty('change')
  })

  it('Rate onChange Event', () => {
    const wrapper = mount(<Range />)
    wrapper.find('.ob-range__slider').trigger('touchmove', { x: 50, touches: [{ clientX: 50 }] })
    wrapper.find('.ob-range__slider').trigger('touchend', { x: 100 })
    wrapper.trigger('click', { x: 100 })
    expect(wrapper.emitted()).toHaveProperty('change')
  })

  it('Rate onChange Event', () => {
    const wrapper = mount(<Range />)
    wrapper.find('.ob-range__slider:last-child').trigger('touchmove', { x: 50, touches: [{ clientX: 50 }] })
    wrapper.find('.ob-range__slider:last-child').trigger('touchend', { x: 100 })
    wrapper.trigger('click', { x: 100 })
    expect(wrapper.emitted()).toHaveProperty('change')
  })
})