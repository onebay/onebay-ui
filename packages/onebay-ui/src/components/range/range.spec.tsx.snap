// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Range Snap render Range -- props blockSize 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style="left: 0%;"
    />
    <div
      class="ob-range__slider"
      style="left: 0%;"
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props className 1`] = `
<div
  class="ob-range test"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props customStyle 1`] = `
<div
  class="ob-range"
  style="color: red;"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props disabled 1`] = `
<div
  class="ob-range ob-range--disabled"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props max 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props min 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props railStyle 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
      style="color: red;"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props sliderStyle 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style="color: red;"
    />
    <div
      class="ob-range__slider"
      style="color: red;"
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props trackStyle 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
      style="color: red;"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render Range -- props value 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;

exports[`Range Snap render initial Range 1`] = `
<div
  class="ob-range"
>
  <div
    class="ob-range__container"
  >
    <div
      class="ob-range__rail"
    />
    <div
      class="ob-range__track"
      style="left: 0%; width: 0%;"
    />
    <div
      class="ob-range__slider"
      style=""
    />
    <div
      class="ob-range__slider"
      style=""
    />
  </div>
</div>
`;
